apps/system/js/software_button_manager.js

```diff
@@ -202,19 +201,16 @@
      * @memberof SoftwareButtonManager.prototype
      */
      resizeAndDispatchEvent: function() {
-       var element = this.element;
-       if (this.enabled) {
-         element.addEventListener('transitionend', function trWait() {
-           element.removeEventListener('transitionend', trWait);
-           // Delay posting the event until the transition is done, otherwise
-           // the screen will resize and the background will be visible.
-           window.dispatchEvent(new Event('software-button-enabled'));
-         });
-         element.classList.add('visible');
-       } else {
-         element.classList.remove('visible');
-         window.dispatchEvent(new Event('software-button-disabled'));
-       }
+        var element = this.element;
+        // status[0] = current status
+        // status[1] = previous status
+        var status = ['enabled', 'disabled'];
+        status = this.enabled ? status : status.reverse();
+
+        element.classList.toggle('visible', !!(this.enabled));
+        this.screenElement.classList.add('software-button-' + status[0]);
+        this.screenElement.classList.remove('software-button-' + status[1]);
+        window.dispatchEvent(new CustomEvent('software-button-' + status[0]));
```

Comment:
This is clever, but it's a little bit hard to read at first. Can we have something like:
```
var previousState = this.enabled ? 'software-button-disabled' : 'software-button-enabled';
var currentState = this.enabled ? 'software-button-enabled' : 'software-button-disabled';
this.screenElement.classList.add(currentState);
this.screenElement.classList.remove(previousState);
window.dispatchEvent(new CustomEvent(currentState));
```
I like the simplification, but the code looks more complex than it is currently.